src/rdb.c

```diff
@@ -1820,8 +1824,8 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
                         return NULL;
                     }
                 } else if (setTypeSize(o) < server.set_max_listpack_entries &&
-                           sdslen(sdsele) <= server.set_max_listpack_value &&
-                           lpSafeToAdd(NULL, sdslen(sdsele)))
+                           maxelelen <= server.set_max_listpack_value &&
+                           lpSafeToAdd(NULL, sumelelen))
```

Comment:
theoretically subelelen should go instead of NULL, and maxelelen as the second argument, but that's not possible (the function takes a listpack pointer.
however, since lpSafeToAdd only cares about the sum, that doesn't make any difference

maybe we should put a comment to avoid any future confusion?